src/inmanta/export.py

```diff
@@ -438,7 +441,30 @@ def deploy_code(self, conn: protocol.SyncClient, tid: uuid.UUID, version: Option
 
         upload_code(conn, tid, version, code_manager)
 
-    def commit_resources(self, version: int, resources: List[Dict[str, str]], metadata: Dict[str, str], model: Dict) -> None:
+    def get_resource_sets(self, resource_set_instances: List[Instance]) -> Dict[str, Optional[str]]:
+        """
+        return a dictonary with as keys resource_ids and as values the name of the resource_set
+        the resource belongs to. return None if no resource_set is defined.
+        """
+        resource_sets: Dict[str, Optional[str]] = {}
+        for resource_set_instance in resource_set_instances:
+            name: str = resource_set_instance.get_attribute("name").get_value()
+            resources_in_set: List[Instance] = resource_set_instance.get_attribute("resources").get_value()
+            for resource_in_set in resources_in_set:
+                resource_id: str = self.resource_mapping[resource_in_set].id.resource_str()
```

Comment:
This can fail. 

It is possible for an entity to be of type 'std::Resource' but not be an actual resource that is exported (i.e. the python half is missing).

If such a resource is in the resource set, this will fail. 

for example

```
std::ResourceSet(name="resource_set_1", resources=[std::Resource()])

implement std::Resource using std::none
```

I would add a testcase to this effect.


I'm unsure how to handle this, I think a warning might be sufficient? 
(some resource are also sometimes a resources, sometimes they are not. A resource can throw a specific exception to be discarded. Not failing hard on this may allow more flexible use of such resources, ....) 

@arnaudsjs wdyt?

Reply:
Yes. I think a warning would be sufficient.